vanetza/security/default_certificate_validator.cpp

```diff
@@ -39,46 +29,105 @@ CertificateValidity DefaultCertificateValidator::check_certificate(const Certifi
-            }
+            start = start_and_end.start_validity;
+            end = start_and_end.end_validity;
 
             ++certificate_time_constraints;
         } else if (type == ValidityRestrictionType::Time_End) {
-            EndValidity end = boost::get<EndValidity>(validity_restriction);
-
-            // check if certificate is outdated
-            auto now = convert_time32(m_time_now);
-            if (now > end) {
-                return CertificateInvalidReason::OFF_TIME_PERIOD;
-            }
+            EndValidity time_end = boost::get<EndValidity>(validity_restriction);
+            end = time_end;
 
             ++certificate_time_constraints;
         } else if (type == ValidityRestrictionType::Time_Start_And_Duration) {
             StartAndDurationValidity start_and_duration = boost::get<StartAndDurationValidity>(validity_restriction);
 
-            // check if certificate is premature or outdated
-            auto now = convert_time32(m_time_now);
-            std::chrono::seconds duration = start_and_duration.duration.to_seconds();
-            auto end = start_and_duration.start_validity + duration.count();
-            if (now < start_and_duration.start_validity || now > end) {
-                return CertificateInvalidReason::OFF_TIME_PERIOD;
-            }
+            start = start_and_duration.start_validity;
+            end = start_and_duration.start_validity + start_and_duration.duration.to_seconds().count();
 
             ++certificate_time_constraints;
         }
     }
 
-    // if not exactly one time constraint is given, we fail instead of considering it valid
-    if (1 != certificate_time_constraints) {
+    return certificate_time_constraints == 1;
+}
+
+bool check_time_consistent(const Certificate& certificate, const Certificate& signer)
+{
+    boost::optional<Time32> certificate_time_start;
+    boost::optional<Time32> certificate_time_end;
+
+    boost::optional<Time32> signer_time_start;
+    boost::optional<Time32> signer_time_end;
+
+    if (!extract_validity_time(certificate, certificate_time_start, certificate_time_end)) {
+        return false;
+    }
+
+    if (!extract_validity_time(signer, signer_time_start, signer_time_end)) {
+        return false;
+    }
+
+    if (signer_time_start && *signer_time_start > *certificate_time_start) {
+        return false;
+    }
+
+    if (signer_time_end && *signer_time_end < *certificate_time_end) {
+        return false;
+    }
+
+    return true;
+}
+
+} // namespace
+
+DefaultCertificateValidator::DefaultCertificateValidator(Backend& backend, const Clock::time_point& time_now,
+        const TrustStore& trust_store, CertificateCache& cert_cache) :
+    m_crypto_backend(backend),
+    m_time_now(time_now),
+    m_trust_store(trust_store),
+    m_cert_cache(cert_cache)
+{
+}
+
+CertificateValidity DefaultCertificateValidator::check_certificate(const Certificate& certificate)
+{
+    return check_certificate(certificate, 10);
```

Comment:
Is "10" just a magic number or is there some rationale behind it?

Reply:
It's a magic number, see comment in the regular PR thread.